services/navigation/navigation.go

```diff
@@ -25,11 +26,25 @@ func init() {
 // Mode describes what mode to operate the service in.
 type Mode uint8
 
+// MapType describes what map the navigation service is operating on.
+type MapType uint8
+
 // The set of known modes.
 const (
 	ModeManual = Mode(iota)
 	ModeWaypoint
 	ModeExplore
+
+	NoMap = MapType(iota)
+	GPSMap
+)
+
+// Available modes for each MapType.
+var (
+	AvailableModesByMapType = map[MapType][]Mode{
```

Comment:
Should this go in builtin.go instead? In case it varies for modular navigation services.

Reply:
That's a good point! Just to confirm, we don't want to move the `Mode` and `MapType` from `navigation.go` as well? We'd have to keep the definitions:

// Mode describes what mode to operate the service in.
type Mode uint8

// MapType describes what map the navigation service is operating on.
type MapType uint8

But do we want to consider the `GPS`, `None` and/or `Manual`, `Waypoint`, `Explore` to be implementation specific?

Reply:
I would definitely keep Mode in navigation.go, since that's part of the API. Since MapType is not part of the API, I would put it in builtin.go.

However, should MapType be part of the API? The RC card will depend on the map_type to know whether to display a map. This means it either needs to look in the config (which won't work for modular navigation services), or we need to add it to the API, e.g. GetMapType.

Reply:
I think it would be a good idea to add it to the api unless we want to consider custom modules for Maptype such as heat maps or wifi signal strength maps. Would you like me to send a rescope with this change out or include this as part of the upcoming SLAM navigation scope?

Reply:
I think a scope update for this makes sense, so that the RC card can call GetMapType. Can you check with Esha first?

Reply:
I'm fine leaving MapType in navigation.go for now.